interface.js

```diff
@@ -0,0 +1,282 @@
+$(document).ready(function() {
+
+  var frame1 = new Frame
+  var frame2 = new Frame
+  var frame3 = new Frame
+  var frame4 = new Frame
+  var frame5 = new Frame
+  var frame6 = new Frame
+  var frame7 = new Frame
+  var frame8 = new Frame
+  var frame9 = new Frame
+  var frame10 = new Frame
+  var frame11 = new Frame
+  var frame12 = new Frame
+
+  var score1 = new Score(frame1,frame2,frame3)
+  var score2 = new Score(frame2,frame3,frame4)
+  var score3 = new Score(frame3,frame4,frame5)
+  var score4 = new Score(frame4,frame5,frame6)
+  var score5 = new Score(frame5,frame6,frame7)
+  var score6 = new Score(frame6,frame7,frame8)
+  var score7 = new Score(frame7,frame8,frame9)
+  var score8 = new Score(frame8,frame9,frame10)
+  var score9 = new Score(frame9,frame10,frame11)
+  var score10 = new Score(frame10,frame11,frame12)
+
+  var scorecard = new Scorecard([score1, score2, score3, score4, score5,
+     score6, score7, score8, score9, score10])
+
+  $('#bonus1').attr('class', 'hidden')
+  $('#bonus2').attr('class', 'hidden')
+
+  askForRoll('#roll1frame1', 0)
+
+  $('#roll1frame1').submit(function() {
+    var roll = $('#roll').val();
+    frame1.roll1(roll);
+    $('#roll1frame1cell').html(roll);
+    updateScores()
+    if (roll == 10) {
+      strike(1)
+    } else {
+      askForRoll('#roll2frame1', roll)
+    }
+  })
+  $('#roll2frame1').submit(function() {
+    var roll = $('#roll').val();
+    frame1.roll2(roll);
+    $('#roll2frame1cell').html(roll);
+    askForRoll('#roll1frame2', 0)
+    updateScores()
+  })
+
+  $('#roll1frame2').submit(function() {
+    var roll = $('#roll').val();
+    frame2.roll1(roll);
+    $('#roll1frame2cell').html(roll);
+    updateScores()
+    if (roll == 10) {
+      strike(2)
+    } else {
+      askForRoll('#roll2frame2', roll)
+    }
+  })
+  $('#roll2frame2').submit(function() {
+    var roll = $('#roll').val();
+    frame2.roll2(roll);
+    $('#roll2frame2cell').html(roll);
+    updateScores()
+    askForRoll('#roll1frame3', 0)
+  })
```

Comment:
The code in this block (and the following blocks) is very similar to the code in https://github.com/makersacademy/bowling-challenge/pull/1189/files#diff-c680b607552f1865398553e1ea8ab421R35-R52

Is there a way to refactor these similar repeated blocks to follow the DRY principle of code?

Reply:
yes of course, I think I my brain was looking for ways to create these functions as a means to DRY rather than extract out for them.